tardis/plasma/tests/test_nlte_solver.py

```diff
@@ -218,3 +226,83 @@ def test_jacobian_matrix(
         [0.0, 1.0, 0.0, 1.0, 2.0, -1.0],
     ]
     assert_almost_equal(actual_jacobian_matrix, desired_jacobian_matrix)
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_data_fname(tardis_ref_path):
+    """
+    File name for the atomic data file used in NTLE ionization solver tests.
+    """
+    atomic_data_fname = os.path.join(
+        tardis_ref_path, "nlte_atom_data", "TestNLTE_He_Ti.h5"
+    )
+
+    atom_data_missing_str = (
+        f"{atomic_data_fname} atomic datafiles " f"does not seem to exist"
+    )
+
+    if not os.path.exists(atomic_data_fname):
+        pytest.exit(atom_data_missing_str)
+
+    return atomic_data_fname
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_dataset(nlte_atomic_data_fname):
+    """
+    Atomic dataset used for NLTE ionization solver tests.
+    """
+    nlte_atomic_data = AtomData.from_hdf(nlte_atomic_data_fname)
+    return nlte_atomic_data
+
+
+data_path = os.path.join("tardis", "io", "tests", "data")
+
+
+@pytest.fixture
+def tardis_model_config_nlte():
+    filename = "tardis_configv1_nlte.yml"
+    return Configuration.from_yaml(os.path.join(data_path, filename))
+
+
+@pytest.fixture
+def nlte_raw_model(tardis_model_config_nlte):
+    return Radial1DModel.from_config(tardis_model_config_nlte)
+
+
+@pytest.fixture
+def nlte_raw_plasma(
+    tardis_model_config_nlte, nlte_raw_model, nlte_atomic_dataset
+):
+    """
+    Plasma assembled with dilution factors set to 1.0.
+    """
+    new_w = np.ones_like(nlte_raw_model.dilution_factor)
+    nlte_raw_model.dilution_factor = new_w
+    plasma = assemble_plasma(
+        tardis_model_config_nlte, nlte_raw_model, nlte_atomic_dataset
+    )
+    return plasma
+
+
+def test_critical_case(nlte_raw_plasma):
```

Comment:
I think it would be good to also test the W=0 case to make sure that the collisions are working properly.

Reply:
done!